app/ui/src/app/integration/import-export/import/integration-import.models.ts

```diff
@@ -0,0 +1,36 @@
+import { BaseReducerModel } from '@syndesis/ui/platform';
+
+/**
+ * Step 1: Integration Upload
+ */
+
+export interface IntegrationImportUploadState extends BaseReducerModel {
+  file?: File;
+  importResults: {
+    integrations?: string[];
+    connections?: string[];
+  };
+  list: IntegrationImportsUploadState;
+
+}
+
+export type IntegrationImportsUploadState = Array<IntegrationImportUploadState>;
+
+
+/**
+ * Step 2: Integration Creation
+ * Submitting the uploaded item to be created.
+ */
+
+export interface IntegrationImportEditState extends BaseReducerModel {
+  file?: File;
+  importResults: {
+    integrations?: string[];
+    connections?: string[];
+  };
+  list: IntegrationImportsEditState;
+}
+
+export type IntegrationImportsEditState = Array<IntegrationImportEditState>;
+
```

Comment:
What is the point of defining two different state interfaces for the same schema? And two types for an interface array with different names?

Reply:
If you want to define an interface that models the import operation (regardless it models an upload or edit operation) just do it once! And strive for interface reusability whenever you can (as in this place).